src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -538,10 +538,18 @@ public function getMainVariants()
     /**
      * @arrayAccess
      *
-     * @return 0|null
+     * @return array|null
      */
     public function getSpecificReferences()
     {
+        // If the product has no combinations then the `specific_references` must be filled in
+        if (!isset($this->product['attributes'])) {
+            $this->product['attributes'][0]['isbn'] = $this->product['isbn'] ?? false;
+            $this->product['attributes'][0]['upc'] = $this->product['upc'] ?? false;
+            $this->product['attributes'][0]['ean13'] = $this->product['ean13'] ?? false;
+            $this->product['attributes'][0]['mpn'] = $this->product['mpn'] ?? false;
+        }
+
         if (isset($this->product['attributes']) && !isset($this->product['cart_quantity'])) {
             $specificReferences = array_slice($this->product['attributes'], 0)[0];
```

Comment:
```suggestion
        if (!isset($this->product['cart_quantity'])) {
            return null;
        }
        
        // If the product has no combinations then the `specific_references` must be filled in
        if (isset($this->product['attributes'])) {
            $specificReferences = array_slice($this->product['attributes'], 0)[0];
        } else {
            $specificReferences = [
                'isbn' => $this->product['isbn'] ?? false,
                'upc' => $this->product['upc'] ?? false,
                'ean13' => $this->product['ean13'] ?? false,
                'mpn' => $this->product['mpn'] ?? false,
            ];
        }           
```

@kpodemski Maybe something like that will be better for everyone?

Reply:
@PierreRambaud yeah, looking good, can you move from there @levyn?

Reply:
@kpodemski Yes, I like it. Just a little mistake, "!" - it is not necessary here